compiler/src/dotty/tools/dotc/core/Phases.scala

```diff
@@ -321,9 +321,9 @@ object Phases {
     def run(using Context): Unit
 
     /** @pre `isRunnable` returns true */
-    def runOn(units: List[CompilationUnit])(using Context): List[CompilationUnit] =
+    def runOn(units: List[CompilationUnit])(using runCtx: Context): List[CompilationUnit] =
       units.map { unit =>
-        val unitCtx = ctx.fresh.setPhase(this.start).setCompilationUnit(unit).withRootImports
+        given unitCtx: Context = runCtx.fresh.setPhase(this.start).setCompilationUnit(unit).withRootImports
         try run(using unitCtx)
         catch case ex: Throwable if !ctx.run.enrichedErrorMessage =>
```

Comment:
it's a bit confusing to use unitCtx and ctx simultaneously to refer to the same thing, I guess this could be just `try run`